### PR TITLE
Add new fields to loads for expert filters.

### PR DESCRIPTION
**Pull request by AAJELLAL**

### CODE REVIEW

**Review by YenguiSeddik — APPROVED** [review]
Code & test: ok

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java [hunk 1]
```diff
@@ -442,35 +446,58 @@ private static Stream<Arguments> provideArgumentsForLoadTest() {
...
                 // VoltageLevel fields
                 Arguments.of(BETWEEN, FieldType.NOMINAL_VOLTAGE, null, Set.of(12.0, 14.0), load, true),
                 Arguments.of(BETWEEN, FieldType.NOMINAL_VOLTAGE, null, Set.of(13.5, 14.0), load, false),
+                // Load fields
+                Arguments.of(BETWEEN, FieldType.P0, null, Set.of(60.0, 80.0), load, true),
+                Arguments.of(BETWEEN, FieldType.P0, null, Set.of(60.0, 70.0), load, false),
+                Arguments.of(BETWEEN, FieldType.Q0, null, Set.of(270.0, 280.0), load, true),
+                Arguments.of(BETWEEN, FieldType.Q0, null, Set.of(100.0, 260.0), load, false),
 
                 // --- EXISTS --- //
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:502]: Lack of tests for EXISTS, note that at the moment we use functional test approach, so need to cover all operators for each field of each equipment 

EXIST for number type is defined at : https://github.com/gridsuite/gridexplore-app/blob/main/src/components/dialogs/filter/expert/expert-filter-utils.ts#L65C1-L73C41

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java [hunk 2]
```diff
@@ -442,35 +446,58 @@ private static Stream<Arguments> provideArgumentsForLoadTest() {
                 // VoltageLevel fields
                 Arguments.of(EQUALS, FieldType.NOMINAL_VOLTAGE, 13.0, null, load, true),
                 Arguments.of(EQUALS, FieldType.NOMINAL_VOLTAGE, 14.0, null, load, false),
+                // Load fields
+                Arguments.of(EQUALS, FieldType.P0, 77.0, null, load, true),
+                Arguments.of(EQUALS, FieldType.Q0, 277.0, null, load, true),
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:451]: Could you change the order as following
Arguments.of(EQUALS, FieldType.P0, 50.0, null, load, false),
Arguments.of(EQUALS, FieldType.Q0, 277.0, null, load, true),

We do the test field by field => easier to manage test data

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java [hunk 3]
```diff
@@ -442,35 +446,58 @@ private static Stream<Arguments> provideArgumentsForLoadTest() {
...
+                Arguments.of(EQUALS, FieldType.P0, 50.0, null, load, false),
+                Arguments.of(EQUALS, FieldType.Q0, 300.0, null, load, false),
 
                 // --- GREATER_OR_EQUALS --- //
                 // VoltageLevel fields
                 Arguments.of(GREATER_OR_EQUALS, FieldType.NOMINAL_VOLTAGE, 12.0, null, load, true),
                 Arguments.of(GREATER_OR_EQUALS, FieldType.NOMINAL_VOLTAGE, 13.0, null, load, true),
                 Arguments.of(GREATER_OR_EQUALS, FieldType.NOMINAL_VOLTAGE, 14.0, null, load, false),
+                // Load fields
+                Arguments.of(GREATER_OR_EQUALS, FieldType.P0, 72.0, null, load, true),
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:461]: Each field P0, Q0 possible to add 3 test cases as the order of NOMINAL_VOLTAGE for the homogeneity
For GREATER_OR_EQUALS operator, it is better to test also equal case

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java [hunk 4]
```diff
@@ -442,35 +446,58 @@ private static Stream<Arguments> provideArgumentsForLoadTest() {
...
+                Arguments.of(GREATER_OR_EQUALS, FieldType.Q0, 272.0, null, load, true),
 
                 // --- GREATER --- //
                 // VoltageLevel fields
                 Arguments.of(GREATER, FieldType.NOMINAL_VOLTAGE, 12.0, null, load, true),
                 Arguments.of(GREATER, FieldType.NOMINAL_VOLTAGE, 13.0, null, load, false),
                 Arguments.of(GREATER, FieldType.NOMINAL_VOLTAGE, 14.0, null, load, false),
+                // Load fields
+                Arguments.of(GREATER, FieldType.P0, 70.0, null, load, true),
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:470]: idem as comment for GREATER_OR_EQUALS

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java [hunk 5]
```diff
@@ -442,35 +446,58 @@ private static Stream<Arguments> provideArgumentsForLoadTest() {
...
+                Arguments.of(GREATER, FieldType.Q0, 270.0, null, load, true),
 
                 // --- LOWER_OR_EQUALS --- //
                 // VoltageLevel fields
                 Arguments.of(LOWER_OR_EQUALS, FieldType.NOMINAL_VOLTAGE, 14.0, null, load, true),
                 Arguments.of(LOWER_OR_EQUALS, FieldType.NOMINAL_VOLTAGE, 13.0, null, load, true),
                 Arguments.of(LOWER_OR_EQUALS, FieldType.NOMINAL_VOLTAGE, 12.0, null, load, false),
+                // Load fields
+                Arguments.of(LOWER_OR_EQUALS, FieldType.P0, 80.0, null, load, true),
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:479]: idem as comment for GREATER_OR_EQUALS

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java [hunk 6]
```diff
@@ -442,35 +446,58 @@ private static Stream<Arguments> provideArgumentsForLoadTest() {
...
+                Arguments.of(LOWER_OR_EQUALS, FieldType.Q0, 300.0, null, load, true),
+                Arguments.of(LOWER_OR_EQUALS, FieldType.Q0, 100.0, null, load, false),
 
                 // --- LOWER --- //
                 // VoltageLevel fields
                 Arguments.of(LOWER, FieldType.NOMINAL_VOLTAGE, 14.0, null, load, true),
                 Arguments.of(LOWER, FieldType.NOMINAL_VOLTAGE, 13.0, null, load, false),
                 Arguments.of(LOWER, FieldType.NOMINAL_VOLTAGE, 12.0, null, load, false),
+                // Load fields
+                Arguments.of(LOWER, FieldType.P0, 99.0, null, load, true),
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:489]: idem as comment for GREATER_OR_EQUALS

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java (hunk 1 above)
Comment (AAJELLAL) [reply src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:517]: Done.

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java (hunk 6 above)
Comment (AAJELLAL) [reply src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:500]: Done.

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java (hunk 5 above)
Comment (AAJELLAL) [reply src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:487]: Done.

File: src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java (hunk 3 above)
Comment (AAJELLAL) [reply src/test/java/org/gridsuite/filter/server/expertrule/NumberExpertRuleTest.java:461]: Done.

**Pull request merged**